frontend/src/components/uploaders/CSVFileUploader.tsx

```diff
@@ -4,22 +4,27 @@ const MAX_FILE_UPLOAD = 1000000; // Bytes
 
 interface CSVUploaderProps {
   uploadHandler: (e: FileUploadHandlerEvent) => void;
+  visible: boolean;
   disabled?: boolean;
 }
 
 export default function CSVUploader(props: CSVUploaderProps) {
   return (
-    <div className="card flex justify-content-center">
-      <FileUpload
-        auto
-        mode="basic"
-        accept=".csv"
-        maxFileSize={MAX_FILE_UPLOAD}
-        customUpload
-        disabled={props.disabled ?? false}
-        uploadHandler={props.uploadHandler}
-        chooseLabel={"Import CSV"}
-      />
-    </div>
+    <>
+      {props.visible && (
+        <div className="card flex justify-content-center">
+          <FileUpload
+            auto
+            mode="basic"
+            accept=".csv"
+            maxFileSize={MAX_FILE_UPLOAD}
```

Comment:
This is outside the refactoring comment, but did professor Bletsch say anything about max file size for CSV?

Reply:
Nope, maybe worth asking him in class